packages/x-data-grid/src/components/columnHeaders/GridColumnHeaderTitle.tsx

```diff
@@ -30,6 +30,12 @@ const GridColumnHeaderTitleRoot = styled('div', {
   whiteSpace: 'nowrap',
   fontWeight: 'var(--unstable_DataGrid-headWeight)',
   lineHeight: 'normal',
+  // To prevent Safari adding its own tooltip for truncated text
```

Comment:
This issue is still present on the `GridCell`, but I couldn't apply the same because it would [impact performance](https://github.com/mui/mui-x/pull/12013#discussion_r1486541886)